src/core/utils.js

```diff
@@ -14,4 +14,16 @@ exports.ifRepoExists = (repo, cb) => {
   })
 }
 
+exports.repoExists = (self) => {
+  return (cb) => {
+    self._repo.exists((err, exists) => {
+      if (err) {
+        return cb(err)
+      }
+
+      cb(err, exists)
+    })
+  }
+}
```

Comment:
Let's converge repoExists and ifRepoExists and make the converged function `async` (with callbacks and a promisify wrapper), the reason why it needs to be async is because the repo might be over a wire (like a S3 bucket).